src/utils/updatePackageVersions.js

```diff
@@ -0,0 +1,24 @@
+//@flow
+import Project from '../Project';
+import Workspace from '../Workspace';
+import runWorkspaceTasks from './runWorkspaceTasks';
+
+export default async function updatePackageVersions(versions: {[name: string]: string}, opts: { cwd?: string } = {}) {
+  const cwd = opts.cwd || process.cwd();
+
+  const updateWorkspacePackageVersion = (workspace: Workspace) => {
+    const pkg = workspace.pkg;
+    const promises = [];
+
+    for(let depName of Object.keys(versions)) {
+      const depType = pkg.getDependencyType(depName);
+      if (!depType) continue;
+
+      promises.push(pkg.updateDependencyVersionRange(depName, depType, versions[depName]));
+    }
+
+    return Promise.all(promises);
+  }
+
+  await runWorkspaceTasks(updateWorkspacePackageVersion, opts);
```

Comment:
You don't need `runWorkspaceTasks` for this, it's meant only for scripting.

Reply:
Should this take in a Workspace then?

Reply:
Nah, we should not expose internal objects like `Workspace` or `Package`. This is all you need to do:

```
let cwd = process.cwd();
let project = await Project.init(cwd);
let workspaces = await project.getWorkspaces();

for (let workspace of workspaces) {
  // ...
}
```